### PR TITLE
Add Nonnull annotations

### PR DESCRIPTION
**Pull request by mhabzda**

### CODE REVIEW

File: core/src/main/java/agency/tango/skald/core/SkaldAuthService.java [hunk 1]
```diff
@@ -10,16 +11,17 @@
 import agency.tango.skald.core.provider.ProviderName;
 
 public class SkaldAuthService {
-  private final Context context;
-  private final OnAuthErrorListener onAuthErrorListener;
+  @NonNull private final Context context;
```
Comment (bezmian) [inline core/src/main/java/agency/tango/skald/core/SkaldAuthService.java]: enter on fields

File: core/src/main/java/agency/tango/skald/core/provider/Provider.java [hunk 2]
```diff
@@ -5,6 +5,7 @@
 import agency.tango.skald.core.factories.SkaldAuthStoreFactory;
 import agency.tango.skald.core.factories.UserServiceFactory;
 import agency.tango.skald.core.models.SkaldPlayableEntity;
+import io.reactivex.annotations.NonNull;
```
Comment (radzio) [inline core/src/main/java/agency/tango/skald/core/provider/Provider.java]: why not import `android.support.annotation.NonNull;`

File: deezer/src/main/java/agency/tango/skald/deezer/player/DeezerPlayer.java [hunk 3]
```diff
@@ -23,6 +23,7 @@
 import agency.tango.skald.core.models.SkaldPlayableEntity;
 import agency.tango.skald.core.models.SkaldPlaylist;
 import agency.tango.skald.core.models.SkaldTrack;
+import io.reactivex.annotations.NonNull;
```
Comment (radzio) [inline deezer/src/main/java/agency/tango/skald/deezer/player/DeezerPlayer.java]: why not import `android.support.annotation.NonNull;`

File: deezer/src/main/java/agency/tango/skald/deezer/player/SkaldDeezerPlayer.java [hunk 4]
```diff
@@ -11,6 +11,7 @@
 import agency.tango.skald.core.listeners.OnPlayerReadyListener;
 import agency.tango.skald.core.models.SkaldPlayableEntity;
 import agency.tango.skald.deezer.authentication.DeezerAuthData;
+import io.reactivex.annotations.NonNull;
```
Comment (radzio) [inline deezer/src/main/java/agency/tango/skald/deezer/player/SkaldDeezerPlayer.java]: why not import `android.support.annotation.NonNull;`

**Comment by mhabzda** [comment]
add more annotations
e.g. in models and Skald class

**Pull request merged**